### PR TITLE
Add averaged model and new-bob scheduler to rnnlm 

### PR DESCRIPTION
**Pull request by yfyeung**
- [x] Add NewBob scheduler
- [x] Add support for averaged model
- [x] Fix typo 

### Result on GigaSpeech XL transcription
| method | ppl |
|-- |-- |
| Adam | |
| + NewBobScheduler | |

### CODE REVIEW

File: icefall/rnn_lm/train.py [hunk 1]
```diff
@@ -589,10 +663,20 @@ def run(rank, world_size, args):
         lr=params.lr,
         weight_decay=params.weight_decay,
     )
+    scheduler = NewBobScheduler(optimizer)
```
Comment (marcoyang1998) [inline icefall/rnn_lm/train.py:666]: Could you make it possible for users to pass some arguments to the scheduler?

Comment (yfyeung) [reply icefall/rnn_lm/train.py:666]: Sure.

**Comment by videodanchik** [comment]
Any results improvements with averaged model + Scheduler?

**Comment by danpovey** [comment]
sorry, we are now using very different learning rate schedules and a different optimizer.
we need to revisit the RNNLM work in the near future, possibly using a transformer-based teacher model.
I think this recipe is a little outdated.
